Formula/botan.rb

```diff
@@ -21,6 +21,7 @@ class Botan < Formula
   uses_from_macos "zlib"
 
   def install
+    ENV.prepend_path "PATH", Formula["python@3.9"].opt_libexec/"bin"
```

Comment:
Let's change the `system "./configure"` call to
```ruby
system "python3", "configure.py", *args
```
So that we don't need to change the `@3.9` in two places when we get Python 3.10. We probably also want to change `3.9` in `args` to `#{Language::Python.major_minor_version "python3"}`.

Reply:
It is possible that we can avoid setting the arg as the Python version defaults to running instance of Python. I am guessing this may have been done due to `/usr/bin/env python` picking up Python2 on macOS.
```
    build_group.add_option('--with-python-versions', dest='python_version',
                           metavar='N.M',
                           default='%d.%d' % (sys.version_info[0], sys.version_info[1]),
                           help='where to install botan2.py (def %default)')
```

Reply:
Just checked and it should pick up correct version without overriding in arg:
```console
❯ /opt/homebrew/bin/python3 configure.py --help | grep with-python
    --with-python-versions=N.M           where to install botan2.py (def 3.9)
```

Reply:
Is that their default? or based on the python they find? It'd be preferable if we didn't have to wait for an app update to test python 3.10

Reply:
The python they find, so it should automatically pick up 3.10 when we change dependency to be `depends_on "python@3.10"`.

Reply:
LGTM then